ROX:25471: Update release notes for 4.5.1 patch (#79737)
openshift/openshift-docs: release_notes/45-release-notes.adoc

This release includes the following updates to the *Compliance* view:

* You can integrate with an email server to send scheduled reports.

> * You can now generate an on-demand report for any scan configuration. {product-title-short} can send it by email.

Author: Revised per comment from Maria (PM) via Slack.

* Direct upgrades from {product-title-short} version 3.74 or earlier to version 4.5 are no longer supported. To upgrade to version 4.5 or later, you must first upgrade to version 4.4.

[id="bug-fixes_{context}"]
== Bug fixes in version 4.5.0

*Release date*: 24 July 2024

* Previously, a bug prevented users without administrator privileges from accessing the listening endpoint data for clusters and namespaces for which they had `read` permission. With this update, users with `read` permission can now receive data from the listening endpoints service.

[id="about-release-451_{context}"]
== About release 4.5.1

This release includes notable technical changes and bug fixes.

[id="changes-in-version-451_{context}"]
=== Changes in version 4.5.1

* The network graph and the network policy generator in the {product-title-short} portal were updated to clarify that `AdminNetworkPolicy` and `BaselineAdminNetworkPolicy` resources are not taken into consideration in the network graph or during network policy generation.

> * The field `error` returned for failed API calls has been deprecated, and will be removed in a future release. Instead of using the `error` field, use the `message` field. The `message` field contains the same information as the `error` field.

Reviewer: ```suggestion
* The field `error` returned for failed API calls has been deprecated, and is planned for removal in a future release. Instead of using the `error` field, use the `message` field. The `message` field contains the same information as the `error` field.
```
[Per the SSG](https://redhat-documentation.github.io/supplementary-style-guide/#deprecated-and-removed-features), we should say "plan to remove" rather than a definitive "will be removed" in our docs.

[id="resolved-in-version-451_{context}"]
=== Bug fixes in version 4.5.1

> *Release date:* 8 August 2024

Reviewer: This release date does not match the release date in Table 1 on the top of the preview page. Which date is correct?

> * When upgrading to version 4.5, in some situations, the upgrade failed with a `central-db` error for the `uni_compliance_integrations_clusterid` constraint. This problem has been fixed.

Reviewer: ```suggestion
* When upgrading to {product-title-short} 4.5, in some situations, the upgrade failed with a `central-db` error for the `uni_compliance_integrations_clusterid` constraint. This problem has been fixed.
```
Just a suggestion for clarity.

> * When upgrading to version 4.5 with the {product-title-short} Operator, the upgrade failed with a `central-db` error for the `uni_compliance_integrations_clusterid` constraint. This problem has been fixed.

Reviewer 2: ```suggestion
* When upgrading to version 4.5, the upgrade failed with a `central-db` error for the `uni_compliance_integrations_clusterid` constraint. This problem has been fixed.
```

> * After upgrading to version 4.5.0, for nodes with more than 64 cores, Collector displayed an error about the ring buffer size not being allowed. This problem has been fixed.

Reviewer: ```suggestion
* After upgrading to {product-title-short} 4.5, for nodes with more than 64 cores, Collector displayed an error about the ring buffer size not being allowed. This problem has been fixed.
```
A suggestion the same as above, and consistency in the version numbering. You could pick 4.5 or 4.5.X.

> * After upgrading to version 4.5.0, for nodes with more than 64 cores, Collector displayed an error for the per-CPU ring buffer dimension. This problem has been fixed.

Reviewer 3: It would be more correct to say just "ring buffer size". Technically speaking a ring buffer doesn't have to be per-CPU:
```suggestion
* After upgrading to version 4.5.0, for nodes with more than 64 cores, Collector displayed an error about the not allowed ring buffer size. This problem has been fixed.
```

* Direct upgrades from {product-title-short} version 3.74 or earlier to version 4.5 are no longer supported. To upgrade to version 4.5 or later, you must first upgrade to version 4.4.

[id="bug-fixes_{context}"]
== Bug fixes in version 4.5.0

*Release date*: 24 July 2024

* Previously, a bug prevented users without administrator privileges from accessing the listening endpoint data for clusters and namespaces for which they had `read` permission. With this update, users with `read` permission can now receive data from the listening endpoints service.

[id="about-release-451_{context}"]
== About release 4.5.1

This release includes notable technical changes and bug fixes.

[id="changes-in-version-451_{context}"]
=== Changes in version 4.5.1

* The network graph and the network policy generator in the {product-title-short} portal were updated to clarify that `AdminNetworkPolicy` and `BaselineAdminNetworkPolicy` resources are not taken into consideration in the network graph or during network policy generation.

> * The field `error` returned for failed API calls has been deprecated, and is planned for removal in a future release. Instead of using the `error` field, use the `message` field. The `message` field contains the same information as the `error` field.

Author: Oops, I did not realize! Will remove the duplicate.

> * Scanner was updated to version 2.34.1, which contains a fix for link:https://access.redhat.com/security/cve/cve-2024-41110[CVE-2024-41110]: Vulnerability in authorization plugins in Docker Engine (AuthZ).

Reviewer 4: The same CVE is also fixed in stackrox
